### PR TITLE
7/18

### PR DESCRIPTION
**Pull request by vdabro2**

### CODE REVIEW

File: app/src/main/java/com/example/travellens/MessageActivity.java [hunk 1]
```diff
@@ -63,25 +74,58 @@ protected void onCreate(Bundle savedInstanceState) {
...
     }
 
     private void sendMessage(View view) {
         reference = FirebaseDatabase.getInstance().getReference();
 
+        Date date = new Date();
+        SimpleDateFormat formatter = new SimpleDateFormat("MMM dd hh:mm a");
+        String dateAndTime = formatter.format(date);
+
         HashMap<String, Object> hashMap = new HashMap<>();
```
Comment (daniakou-fb) [inline app/src/main/java/com/example/travellens/MessageActivity.java:114]: Should use a more descriptive name than just hashMap.

File: app/src/main/java/com/example/travellens/MessageActivity.java [hunk 2]
```diff
@@ -63,25 +74,58 @@ protected void onCreate(Bundle savedInstanceState) {
...
         hashMap.put("sender", ParseUser.getCurrentUser().getString(Post.KEY_FIREBASE_USER_ID));
         hashMap.put("receiver", post.getUser().getString(Post.KEY_FIREBASE_USER_ID));
         hashMap.put("message", etMessage.getText().toString());
+        hashMap.put("dateAndTime", dateAndTime);
+        if (post != null && attachPhotoToMessage == true) {
+            // attach image to message
+            hashMap.put("photo", post.getParseFile(Post.KEY_IMAGE).getUrl());
+        } else {
+            hashMap.put("photo", null);
```
Comment (daniakou-fb) [inline app/src/main/java/com/example/travellens/MessageActivity.java:123]: Do you need this put with a null value while adding the message?

**Review by daniakou-fb — APPROVED** [review]

File: app/src/main/java/com/example/travellens/MessageActivity.java (hunk 2 above)
Comment (vdabro2) [reply app/src/main/java/com/example/travellens/MessageActivity.java:123]: I don't think I necessarily have to, but I did for code readability for when the message does not have an attached photo. I'll test it out and remove it if I can.

Comment (daniakou-fb) [reply app/src/main/java/com/example/travellens/MessageActivity.java:123]: Actually even for code readability, it's not really needed. The if condition itself already lets us know that the photo is only added to the map if there is a photo available to add.

**Pull request merged**